src/relo_core.c

```diff
@@ -1732,7 +1847,7 @@ int bpf_core_apply_relo_insn(const char *prog_name, struct bpf_insn *insn,
 	}
 
 patch_insn:
-	if (reloc_info) {
+	if (reloc_info && targ_res.targ_spec) {
```

Comment:
```suggestion
	if (reloc_info && targ_res.validate && targ_res.targ_spec) {
```

since targ_spec is a pointer the compiler has no guarantees it will be able to find targ_res initialized after a goto because targ_res itself is a raw value.

There are two ways of solving it
-  checking a scalar value that has a default value (like `validate` in this case)
- make targ_res a pointer

I sent a suggestion here with the first solution

Reply:
What about doing `targ_res.targ_spec = NULL;` just before `goto patch_insn;`?

Reply:
Yep, I thought the same (on the = NULL) when I first read @fntlnz's comment. I think its cleaner.